tests/unit/specification/parser_test.py

```diff
@@ -1925,6 +1925,203 @@ def test_parse_error_invalid_context_clause(tmp_path: Path) -> None:
+               end Session;
+
+            end Test;
+        """,
+        r'^<stdin>:11:62: parser: error: duplicate aspect "Writable"',
+    )
+
+
+def test_parse_error_duplicate_checksum_aspect() -> None:
+    assert_error_string(
+        """
+            package Test is
+               type T is mod 2**8;
+               type M is
+                  message
+                     F1 : T;
+                     F2 : T;
+                     C1 : T;
+                     F3 : T;
+                     C2 : T
+                        then F4
+                           if C1'Valid_Checksum and C2'Valid_Checksum;
+                     F4 : T;
+                  end message
+                     with Checksum => (C1 => (F3, F1'First .. F2'Last),
+                                       C2 => (C1'Last + 1 .. C2'First - 1)),
+                          Checksum => (C1 => (F3, F1'First .. F2'Last),
+                                       C2 => (C1'Last + 1 .. C2'First - 1)),
+                          Byte_Order => Low_Order_First;
+            end Test;
+        """,
+        # ISSUE: Componolit/RecordFlux-parser#56
+        # Delete the following line when ticket mentioned above is solved.
+        "^<stdin>:17:27: parser: error: Expected 'Byte_Order', got 'First'",
+        # Uncomment this when ticket #56 is solved.
+        #        r'^<stdin>:17:27: parser: error: duplicate aspect "Checksum"',
+    )
+
+
+def test_parse_error_duplicate_byte_order_aspect() -> None:
+    assert_error_string(
+        """
+            package Test is
+               type T is mod 2**8;
+               type M is
+                  message
+                     F1 : T;
+                     F2 : T;
+                     C1 : T;
+                     F3 : T;
+                     C2 : T
+                        then F4
+                           if C1'Valid_Checksum and C2'Valid_Checksum;
+                     F4 : T;
+                  end message
+                     with Checksum => (C1 => (F3, F1'First .. F2'Last),
+                                       C2 => (C1'Last + 1 .. C2'First - 1)),
+                          Byte_Order => Low_Order_First,
+                          Byte_Order => Low_Order_First;
+            end Test;
+        """,
+        r'^<stdin>:18:27: parser: error: duplicate aspect "ByteOrderAspect"',
+    )
+
+
+def test_parse_error_duplicate_always_valid_aspect() -> None:
+    assert_error_string(
+        """
+            package Test is
+               type T is (A, B) with Always_Valid => True, Always_Valid => False;
+            end Test;
+        """,
+        r'^<stdin>:3:60: parser: error: duplicate aspect "Always_Valid"',
+    )
+
+
+def test_parse_error_duplicate_size_aspect() -> None:
+    assert_error_string(
+        """
+            package Test is
+               type T is (A, B) with Size => 12, Size => 10;
+            end Test;
+        """,
+        r'^<stdin>:3:50: parser: error: duplicate aspect "Size"',
+    )
```

Comment:
Are duplicate Size aspects also detected for range integers?

Reply:
The logic of detecting duplicate Size aspects doesn't care about the format of the expression given to the aspect Size, so the answer to your question is Yes.

Reply:
Please add a regression test nevertheless.

Reply:
wouldn't be code duplication as well?